include/kernel.h

```diff
@@ -504,6 +504,13 @@ extern void k_call_stacks_analyze(void);
  */
 #define K_USER (1 << 2)
 
+/* This thread should make its stack memory permissions universally accessible
+ * if and when the thread aborts. Allows for re-use of stack memory areas
+ * if CONFIG_USERSPACE is enabled. Only use this with K_USER, and if you are
+ * sure that the released stack memory will not have any sensitive information.
+ */
+#define K_STACK_RELEASE (1 << 3)
```

Comment:
Why not define K_STACK_RELEASE as `(K_USER | 1 << 3)`?  By modifying `_x86_swap_update_page_tables()` slightly, this will enforce the need for this to be used only with user threads.

Reply:
I actually have a use-case for using this without K_USER, in a test suite that reuses the same stack/thread object over and over, but some tests are in user mode and some aren't.

I think I can update the comment and remove that verbiage.